### DISCUSSION TITLE
fetch requests to Base lead to an internal server error

### DISCUSSION THREAD

**Discussion by uchokoro** [deta/cloud-docs#367] [Help]
In fetching data from Base,  get an **internal server error** on my micro once the limit value exceeds the number of items/records/document in the database matched by the query minus 1. Even when I simply use **fetch()**, implying a limiit value of 1000 - the default, this occurs.

What could be the cause of this, and how do I resolve it?

My database currentl has less than 20 documents in it.

**Comment by abdelhai** [comment]
maybe this error is not related to base... did you check your error logs?

**Reply by uchokoro** [reply]
Funnily, Visor on my Micro does not log any error. Rather it logs a response with the message "Internal server error". 

Everything works beautifully, as long as the limit is set to a value less than the total number of items matched by the query to Base. However, once the limit becomes equal to, or greater than, the said number of items, I get this 'error'. I have tried to catch a broad range of exceptions in my code, including  BaseException, to help me identify the exact cause of the issue. However, my exception catching code does not get triggered. My thinking is that whatever error is being triggered get's caught somewhere down the stack, and the message "Internal server error" is returned as the HTTP response.

**Reply by abdelhai** [reply]
try to run `deta logs --follow` and run a couple requests and see what messages you get in the terminal.

**Reply by uchokoro** [reply]
Thanks a lot @abdelhai. I've been able to identify that the issue arises from my not adequate provision in my data models for situations in which the value of the last property in the return FetchResponse object is null.
The issue is now resolved.
Thanks, once again.
